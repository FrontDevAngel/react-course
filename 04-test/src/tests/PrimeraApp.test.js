import React from 'react';
import'@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Prueba en componente PrimeraApp', () => {
    // test('Debe de mostrar el mensaje: Hola, soy Ángel', () => {
    //     const saludo = 'Hola, soy Ángel';

    //     const { getByText } = render( <PrimeraApp saludo={saludo} /> );

    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe de mostrar el mensaje: Hola, soy Ángel', () => {
        const saludo = 'Hola, soy Ángel';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola, soy Ángel';
        const subtitulo = 'Front End Dev';

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo} 
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);

    })  
})