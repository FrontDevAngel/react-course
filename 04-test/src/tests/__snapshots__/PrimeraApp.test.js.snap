// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente PrimeraApp Debe de mostrar el mensaje: Hola, soy Ángel 1`] = `
<Fragment>
  <h1>
    Hola, soy Ángel
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
